ledger/ledger.go

```diff
@@ -244,9 +241,11 @@ func initBlocksDB(tx *sql.Tx, l *Ledger, initBlocks []bookkeeping.Block, isArchi
 // and goroutines used by trackers).
 func (l *Ledger) Close() {
 	l.trackerDBs.close()
-	l.blockQ.close()
 	l.blockDBs.close()
 	l.trackers.close()
+	if l.blockQ != nil {
+		l.blockQ.close()
```

Comment:
nit: if you're already testing for blockQ being nil, then you might consider clearing it out after ( or even better - before ) calling the close.

Reply:
Done